build-logic/documentation/src/main/groovy/gradlebuild/docs/FindBrokenInternalLinks.java

```diff
@@ -151,6 +128,55 @@ private void hasDeadLink(File baseDir, File sourceFile, Map<File, List<Error>> e
         }
     }
 
+    private void gatherDeadLinksInLine(File sourceFile, String line, int lineNumber, List<Error> errorsForFile) {
+        Matcher matcher = linkPattern.matcher(line);
+        while (matcher.find()) {
+            MatchResult xrefMatcher = matcher.toMatchResult();
+            String link = xrefMatcher.group(1);
+            if (link.contains("#")) {
+                Matcher linkMatcher = linkWithHashPattern.matcher(link);
+                if (linkMatcher.matches()) {
+                    MatchResult result = linkMatcher.toMatchResult();
+                    String fileName = getFileName(result.group(1), sourceFile);
+                    File referencedFile = new File(getDocumentationRoot().get().getAsFile(), fileName);
+                    if (!referencedFile.exists() || referencedFile.isDirectory()) {
+                        errorsForFile.add(new Error(lineNumber, line, "Looking for file named " + fileName));
+                    } else {
+                        String idName = result.group(2);
+                        if (idName.isEmpty()) {
+                            errorsForFile.add(new Error(lineNumber, line, "Missing section reference for link to " + fileName));
+                        } else {
+                            if (!fileContainsText(referencedFile, "[[" + idName + "]]")) {
+                                errorsForFile.add(new Error(lineNumber, line, "Looking for section named " + idName + " in " + fileName));
+                            }
+                        }
+                    }
+                }
+            } else {
+                if (!fileContainsText(sourceFile, "[[" + link + "]]")) {
+                    errorsForFile.add(new Error(lineNumber, line, "Looking for section named " + link + " in " + sourceFile.getName()));
+                }
+            }
+        }
+    }
+
+    private void gatherDeadJavadocLinksInLine(File sourceFile, String line, int lineNumber, List<Error> errorsForFile) {
+        Matcher matcher = javadocLinkPattern.matcher(line);
+        while (matcher.find()) {
+            MatchResult linkMatcher = matcher.toMatchResult();
+            String link = linkMatcher.group(1);
+            File referencedFile = new File(getJavadocRoot().get().getAsFile(), link);
+            if (!referencedFile.exists() || referencedFile.isDirectory()) {
+                String errMsg = "Missing Javadoc file for " + link + " in " + sourceFile.getName();
+                if (link.startsWith("javadoc")) {
+                    errMsg += " (Did you mean to include the leading `javadoc` path component?)";
```

Comment:
This check seems pretty arbitrary now. I get that we had existing examples which included the `javadoc` component, but now that we have this check, how often are we going to run into this case? 

Though, if we leave this in, I think the message wording is confusing. It almost feels like its suggesting to add the path component. Consider updating the text to suggest removing the component. `Did you mean to include` sounds like its suggesting that I include the component in the URL.

Reply:
This check is primarily for me. I've done this like 5 times. 🤭

Good point on the message, I'll update.